### ISSUE TITLE
Function re-naming proposals

### ISSUE THREAD

**Issue by patrickbarks** [jonesor/Rcompadre#45]
__(1)__ Change verb in `cleanDB`?
'Clean' to me implies that something will be changed or removed, but `cleanDB` just adds columns flagging potential issues with the MPMs. I suggest changing to something like `flagDB` (or perhaps something more verbose, like `flagMatrixIssues`).

__(2)__ Change verb in `mergeDBs`?
`mergeDBs` is essentially `rbind.data.frame`. We have a separate CompadreDB method for the base R `merge` function (which performs a join operation), so I suggest changing the verb in `mergeDBs` to 'bind' or 'rbind', or moving this functionality to an `rbind.CompadreDB` method.

__(3)__ Switch to snake_case for all non-accessor functions?
To avoid the awkwardness of camelCase with acronyms ('cleanDB' or 'cleanDb'?), and the mild ambiguity of names like 'DBToFlat' (is the object 'DB' or 'DBT'?).

I think the object_verb format ([recommended by rOpenSci](https://ropensci.github.io/dev_guide/building.html#function-and-argument-naming)) would work really well in Rcompadre. I'd suggest switching the object from `db_` to `cdb_` (COMPADRE Database) because dplyr has a bunch of `db_` functions. E.g.

```
fetchDB         ->  cdb_fetch
cleanDB         ->  cdb_flag
compareDBs      ->  cdb_compare (or cdbs_compare)
mergeDBs        ->  cdb_rbind (or cdbs_rbind, or rbind method)
DBToFlat        ->  cdb_flatten
checkSpecies    ->  cdb_check_species (or cdb_species; or remove)
getMeanMatF     ->  cdb_get_mean_F (or mean_mat_F)
asCompadreDB    ->  as_cdb
convertLegacyDB ->  (remove and just use as_cdb)
stringToMatrix  ->  string_to_mat
```
If we make that change, we could consider changing the 'CompadreDB' class to 'cdb', and then for consistency, the 'CompadreMat' class to something like 'cmat' (COMPADRE Matrix) or 'cmpm' (COMPADRE Matrix Population Model). We could alternatively use a longer object name to avoid potential collisions with other packages, like 'compdb' or 'compadre'. Or keep the current class names and use the `cdb_` convention for functions anyway.

Thoughts?

**Comment by jonesor** [comment]
I like these suggestions. The name for `cleanDB` has been bugging me!

I think that your rOpenSci recommendations are great. I was thinking we could submit an application note about the package to Methods in Ecology and Evolution and they have a collaboration with rOpenSci to [review R packages for publication](https://ropensci.org/blog/2017/11/29/review-collaboration-mee/), so it makes sense to follow their guidelines.

**Comment by patrickbarks** [comment]
closed by #48
